### PR TITLE
Convert shared subcommands to apps

### PR DESCRIPTION
**Pull request by djmitche**
### What does this PR do?

Converts commands owned by agent-shared-components to Fx Apps.

### Motivation

Top-down approach to componentization.

### Additional Notes

This PR is against a base branch containing #13699 and #13662.  Once those are merged, I'll change the base of this PR to `main` and rebase, but it can be reviewed in the interim.

### Describe how to test/QA your changes

Compare the output of the following commands in the previous and current versions of the agent.

#### `agent version`

 * `agent version`
 * `agent version | cat` (so stdout is not a tty)
 * `agent version -h`
 * `agent version -n`
 * `agent version --no-color=false`

in particular, compare when color is and is not present.

#### `agent troubleshooting`

Set up a running agent with
```
inventories_min_interval: 1
inventories_max_interval: 10
```

* `agent troubleshooting`
* `agent troubleshooting -h`
* `agent troubleshooting metadata_v5` (some diffs are OK in the resulting JSON)
* `agent troubleshooting metadata_v5 -h`
* `agent troubleshooting metadata_inventory` (output should match)
* `agent troubleshooting metadata_inventory -h`

#### `agent stop`

* `agent stop -h`
* `agent stop` (with a running agent)

#### `agent status`

You'll need a running agent.

* `agent status -h`
* `agent status`
* `agent status | cat` (so stdout is not a tty, and there is no color)
* `agent status -n`
* `agent status -j`
* `agent status -p`
* `agent status component xyz` (should error - no such thing)
* `agent status component py`

#### agent secret-helper

 * `agent secret-helper`
 * `agent secret-helper -h`
 * `agent secret-helper read -h`
 * `echo '{"version": "1.0", "secrets": ["secret1", "secret2"]}' | agent secret-helper read` (should fail)

#### agent secret

Build the [example script for secrets](https://docs.datadoghq.com/agent/guide/secrets-management/?tab=linux#example-executable) and put its path in the `secret_backend_command` config parameter.  Make sure to `chmod go-rwx` the binary.  Start the agent.

* `agent secret`
* `agent secret -h`

#### agent import

Set up:
 * `mkdir /tmp/{oldconfig,newconfig}`
 * `echo '[Main]' > /tmp/oldconfig/datadog.conf`

Compare:

* `agent import`
* `agent import -h`
* `agent import /tmp/oldconfig /tmp/newconfig`
* `agent import -f /tmp/oldconfig /tmp/newconfig`
* `agent import -f /tmp/oldconfig /tmp/newconfig | cat` (to ensure stdout is not a tty, so no color)
* `agent import -n /tmp/oldconfig /tmp/newconfig`

#### agent hostname

* `agent hostname`
* `agent hostname -h`

#### agent health

* `agent health -h`
* `agent health`
* `agent health | cat` (so stdout is not a tty)
* `agent health -n`

#### agent flare

Don't actually send a flare.  We're concerned with command-line parsing here.

* `agent flare -h`
* `agent flare`
* `agent flare -e notvalid@example.com -l`
* `agent flare 99999 -e notvalid@example.com`
* `agent flare 99999 -e notvalid@example.com | cat` (shouldn't be color in output)
* `agent flare 99999 -n -e notvalid@example.com`
* `agent flare 99999 -e notvalid@example.com -p 30`
* `agent flare 99999 -e notvalid@example.com -p 30 -B -M`

#### agent diagnose

* `agent diagnose -h`
* `agent diagnose`
* `agent diagnose -n`
* `agent diagnose | cat`
* `agent diagnose datadog-connectivity -h`
* `agent diagnose datadog-connectivity`
* `agent diagnose metadata-availability -h`
* `agent diagnose metadata-availability`

#### agent configcheck

* `agent configcheck -h`
* `agent checkconfig -h`
* `agent configcheck -n`
* `agent configcheck | cat`
* `agent configcheck -v` (yep, it's the same as without -v

#### agent check

* `agent check -h`
* `agent check uptime`
* `agent check -n uptime`
* `agent check uptime | cat` (should not have color output)
* `agent check uptime -r`
* `agent check uptime -t 3`
* `agent check uptime --flare`
* `agent check uptime --full-sketches` (note that there are no sketches here so this doesn't look different)
* `agent check uptime --discovery-min-instances 3 --discovery-timeout 20` (verify it takes 20s to timeout, and reports the one instance found)
* `agent check uptime --json`
* `agent check uptime -t 3 --pause 2000` (verify it pauses for 6 seconds total)
* `agent check uptime --table`

#### agent config

* `agent config`
* `agent config -h`
* `agent config list-runtime` (order is not important)
* `agent config get log_level`
* `agent config set log_level debug` (or something it's not already set to)
* `agent config get log_level`

#### agent launch-gui

Run an agent with `gui_port` set (it's disabled by default on linux).

* `agent launch-gui`
* `agent launch-gui -h`

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by djmitche** [comment]
TestGatewayLookupSubnetLookupError in dd-gitlab/kitchen_test_system_probe_linux_x64: [ubuntu, ubuntu-16-04-4.4,ubuntu-18-04,.. is a known intermittent and can be ignored.

File: cmd/agent/subcommands/secret/command.go [hunk 1]
```diff
@@ -12,56 +12,67 @@ import (
...
+		fmt.Println(err)
+		return nil
+	}
+	return nil
+}
+
+func showSecretInfo(config config.Component) error {
 	c := util.GetClient(false)
-	ipcAddress, err := config.GetIPCAddress()
+	ipcAddress, err := pkgconfig.GetIPCAddress()
```
Comment (hush-hush) [inline cmd/agent/subcommands/secret/command.go:71]: Do we plan to move this to the config interface in a later PR ?

**Review by hush-hush — APPROVED** [review]

File: cmd/agent/subcommands/secret/command.go (hunk 1 above)
Comment (djmitche) [reply cmd/agent/subcommands/secret/command.go:71]: I'm not sure -- there's a _lot_ of purpose-specific stuff in `pkg/config` and I'd rather not add all of that to comp/core/config.  I can think of a few ways to address this but haven't made any decisions yet.

**Review by ogaca-dd — APPROVED** [review]

File: cmd/agent/subcommands/config/command.go [hunk 2]
```diff
@@ -89,19 +93,29 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
...
-			}
-			return getConfigValue(getClient, cmd, args)
+			cliParams.args = args
+			return fxutil.OneShot(getConfigValue,
+				fx.Supply(cliParams),
+				fx.Supply(core.BundleParams{
+					ConfFilePath:      globalParams.ConfFilePath,
+					ConfigLoadSecrets: false,
+				}.LogForOneShot("CORE", "off", true)),
+				core.Bundle,
```
[the remark this answers is not in the document]
Comment (djmitche) [reply cmd/agent/subcommands/config/command.go:103]: I did a similar thing [here](https://github.com/DataDog/datadog-agent/pull/13722/commits/4ab611e9d0dcb8fb61d7e9fb06d966d0e864a688).  I'm not 100% convinced that's clearer, though.  WDYT?

[the remark this answers is not in the document]
Comment (djmitche) [reply cmd/agent/subcommands/config/command.go]: I did this in 14da8fc827e445627a08e74937bd150a2d4abb65

**Pull request merged**